openapi.yaml

```diff
@@ -5087,10 +5087,13 @@ components:
           minimum: 1
           nullable: true
           type: integer
-        environment:
-          description: Target environment for the deployment events, impacts `events`
+        environments:
+          description: Target environments for the deployment events, impacts `events`
             and `stages`.
-          type: string
+          items:
+            type: string
+          minItems: 1
```

Comment:
I understand but this is not how it's going to work with our customers. We can fairly assume that if one customer is sending us deploy data it's because they want to see it. This assumption will stand till next year.

I'm just afraid of the use case where one customer will use unique identifiers in the env names. And it would be nice in that case to show the PRs deployed in those envs, without passing an infinite list of envs to the webapp.

Reply:
I mean passing the list of all envs would not be the biggest problem anyway. We'll have bigger issues in the UI if we met such customers, sending us deploy data for many different envs

Reply:
We should never have these customers, because they must use the deployment tags instead of sending different environments. And we already support filtering by deployment tags underneath.

BTW: we will still return all the deployments info of the PRs in the overview table, if that is what you mean. I am talking about the deployment section where you must filter PRs by the `deployed` event.

We already allowed empty lists in some API filters, and it led to much usage confusion and unexpected server overloads in the past. I want to learn the past mistakes and avoid those.

How about this: not sending `environments` means the `deployed` event/stage matches any deployment in any environment? It should do the job, right?

Reply:
Yes that works. Alright got your reasoning around deploy tags.

Reply:
Let's just mention what you've just described somewhere in the doc.